plugins/main/windows/win-overlay/win-overlay_windows.go

```diff
@@ -140,15 +232,34 @@ func cmdAdd(args *skel.CmdArgs) error {
 		}
 	}()
 	if err != nil {
-		return errors.Annotatef(err, "error while AddHnsEndpoint(%v,%v,%v)", epName, hnsNetwork.Id, args.ContainerID)
+		return nil, errors.Annotatef(err, "error while AddHnsEndpoint(%v,%v,%v)", epName, hnsNetwork.Id, args.ContainerID)
 	}
 
 	result, err := hns.ConstructHnsResult(hnsNetwork, hnsEndpoint)
 	if err != nil {
-		return errors.Annotatef(err, "error while constructResult")
+		return nil, errors.Annotatef(err, "error while constructResult")
 	}
 
 	success = true
+	return result, nil
+}
+func cmdAdd(args *skel.CmdArgs) error {
+	n, cniVersion, err := loadNetConf(args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	var result *current.Result
+	if n.ApiVersion == 2 {
```

Comment:
I am not 100% familiar with HNS/HCN, does HCN only work with API version 2? What about v1?

Reply:
yes that is my understanding from the [hcsshim code](https://github.com/microsoft/hcsshim/blob/master/hcn/hcn.go#L15) and the [MSFT docs](https://docs.microsoft.com/en-us/windows-server/networking/technologies/hcn/hcn-top). HNS works with V1. This PR is creating a separation between both workflows.